### ISSUE TITLE
Just can't find a working example in Ionic 2.0.0-RC0

### ISSUE THREAD

**Issue by behnamazimi** [arielfaur/ionic-audio#78]
Hi,
I can't set providers and I can't match my project by documentation.
Is it possible to give a working example of ionic-audio in RC0?

**Comment by jdell64** [comment]
https://github.com/arielfaur/ionic-audio/issues/77 related?

**Comment by behnamazimi** [comment]
Same as #77 and some other vague points.
A simple example in RC0 would help.

**Comment by biemond** [comment]
+1

**Comment by pascalgermain** [comment]
+1

**Comment by arielfaur** [comment]
Hello everyone! I have updated ionic-audio to RC2. I have to say it was quite challenging with the new ```@NgModule```. While it makes development simpler, there is a complexity behind it and documentation is not updated especially for module developers. 
Anyways, there is a working version ````2.1.2``` already available on npm including a demo project showing how to configure an app with ```@NgModule```. I look forward to your feedback!

**Comment by marteb** [comment]
I just tried the demo project and it works without problems in the browser, but when i try to build it for a device, i get the following errors:

`[12:53:45]  Error: Error at
            /Users/username/IonicProjects/ionic-audio/demo/.tmp/app/ionic-audio/ionic-audio-track-play-component.ngfactory.ts:216:29
[12:53:45]  Supplied parameters do not match any signature of call target.
[12:53:45]  Error at
            /Users/username/IonicProjects/ionic-audio/demo/.tmp/app/ionic-audio/ionic-audio-track-progress-component.ngfactory.ts:397:41
[12:53:45]  Property '_range' is private and only accessible within class 'AudioTrackProgressBarComponent'.
[12:53:45]  Error at
            /Users/username/IonicProjects/ionic-audio/demo/.tmp/app/ionic-audio/ionic-audio-track-progress-component.ngfactory.ts:401:41
[12:53:45]  Property '_showProgress' is private and only accessible within class 'AudioTrackProgressBarComponent'.
[12:53:45]  Error at
            /Users/username/IonicProjects/ionic-audio/demo/.tmp/app/ionic-audio/ionic-audio-track-progress-component.ngfactory.ts:404:41
[12:53:45]  Property '_showDuration' is private and only accessible within class 'AudioTrackProgressBarComponent'.
[12:53:45]  Error at
            /Users/username/IonicProjects/ionic-audio/demo/.tmp/app/ionic-audio/ionic-audio-track-progress-component.ngfactory.ts:462:43
[12:53:45]  Property '_range' is private and only accessible within class 'AudioTrackProgressBarComponent'.
[12:53:45]  Error at /Users/username/IonicProjects/ionic-audio/demo/.tmp/app/app.module.ngfactory.ts:401:82
[12:53:45]  Value of type 'typeof AppModule' is not callable. Did you mean to include 'new'?
[12:53:45]  ngc failed
[12:53:45]  ionic-app-script task: "build"
[12:53:45]  Error: Error`

**Comment by arielfaur** [comment]
@marteb Just fixed all DI issues and lint warnings. Latest version is ```2.1.4```.  Could you give it a try and let me know?

**Comment by marteb** [comment]
Seems to work! Thank you :)

**Comment by marteb** [comment]
It works for me on iOS and in the Browser, but not on Android. On Android i get this error:

`Using Cordova Media provider
main.js:12Loading track assets/audio/filename.mp3
main.js:12Audio error => track assets/audio/filename.mp3
Object
code: 1
__proto__: Object__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Object()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()propertyIsEnumerable: propertyIsEnumerable()toLocaleString: toLocaleString()toString: toString()valueOf: valueOf()get __proto__: __proto__()set __proto__: __proto__()`

**Comment by arielfaur** [comment]
hmmm...interesting, it works on my Nexus device running Android 6. What version are you running?

**Comment by marteb** [comment]
This is on a Nexus 5X with Android N (7).

**Comment by biemond** [comment]
Somehow I had to copy your demo ionic-audio scripts from your module to my own app folder

When I load it from the dist it cannot find 
import { IonicModule } from 'ionic-angular';
in 
https://github.com/arielfaur/ionic-audio/blob/2.0/dist/ionic-audio/ionic-audio.module.ts#L4

**Comment by arielfaur** [comment]
@biemond I experienced exactly the same while developing the module. At the moment it is poorly documented and I cannot find any resources that explain how to properly import an external  .module.ts file from another NPM module. I guess in the meantime we'll have to copy the scripts into the app folder.

**Comment by biemond** [comment]
Ok good to know this. There are so many change from ionic 2 beta to rc,it is almost a new version. probably they should use a different version number

thanks for the hard work 👍

**Comment by marteb** [comment]
Yes, i also copied them because i didn’t find another solution…

**Comment by biemond** [comment]
so it is working perfectly on android but with ios it is not loading, i have to go away from app and come back, then it will initialize.
on ios I am using xcode 8.1 and ios 10.1, on my ipad.
how i can force using WebAudioProvider(It works in ios also), in factory for ios only, because for android cordova-media-plugin works perfectly.

thanks

**Comment by arielfaur** [comment]
@marteb says it works on iOS and not on Android, and @biemond says it works perfectly on Android but not on iOS. What am I supposed to do? 😆

**Comment by arielfaur** [comment]
@biemond You can force using a specific provider changing your app.module.ts file file on the following line: https://github.com/arielfaur/ionic-audio/blob/2.0/demo/src/app/app.module.ts#L31
Replace it with
```
providers: [WebAudioProvider]
```

Don't forget to add the WebAudioProvider import at the top.

**Comment by biemond** [comment]
Awesome, 

Can I check for the platform in the provider, default behavior ( media plugin)  is working fine for android .

**Comment by arielfaur** [comment]
@biemond The example I have earlier is not complete. However, if you want to check for the platform at runtime you will need to use a factory function to create a provider instance according to some runtime variables. Take a look at the ```audioProviderfactory``` included with the module here:
https://github.com/arielfaur/ionic-audio/blob/2.0/demo/src/app/ionic-audio/ionic-audio-providers.ts#L15

You can write your own factory function and use it on your app module replace the same line I mentioned before: 
https://github.com/arielfaur/ionic-audio/blob/2.0/demo/src/app/app.module.ts#L31

So basically you would need to write a function like:
```
function myAudioProviderfactory() {
   if (this.platform.is('android')) {
       return new CordovaMediaProvider() 
   else
       return WebAudioProvider();
}
```

And in your app module:
```
providers: [ { provide: AudioProvider, useFactory: myAudioProviderfactory}]
```

**Comment by biemond** [comment]
Nice 

I got it working ( WebAudioProvider on ipad ) with "ionic platform add ios@4.2.1"
with the latest 4.3.0 even platform and all cordova plugin were hanging somehow.

**Comment by arielfaur** [comment]
@marteb Are still seeing issues on Android?
@biemond  Are you on Android N too? Or still on Android 6?

**Comment by biemond** [comment]
It is working fine on my  nexus 5x (Android N or 7)  also on android 6 HTC one m8

**Comment by marteb** [comment]
@arielfaur no, i tried it today with your latest commit (00ed001) and now it works also with local files. Maybe i just got something wrong the last time… nice work!
